frontend/rpm/handle_rpm.go

```diff
@@ -0,0 +1,35 @@
+package rpm
+
+import (
+	"github.com/Azure/dalec"
+	"github.com/moby/buildkit/client/llb"
+)
+
+// Builds an RPM and source RPM from a spec
+//
+// `topDir` is the rpmbuild top directory which should contain the SOURCES and SPECS directories along with any other necessary files
+//
+// `workerImg` is the image to use for the build
+// It is expected to have rpmbuild and any other necessary build dependencies installed
+//
+// `specPath` is the path to the spec file to build relative to `topDir`
+func Build(topDir, workerImg llb.State, specPath string, opts ...llb.ConstraintsOpt) llb.State {
+	return workerImg.Run(
+		// some notes on these args:
+		//  - _topdir is the directory where rpmbuild will look for SOURCES and SPECS
+		//  - _srcrpmdir is the directory where rpmbuild will put the source RPM
+		//  - _rpmdir is the directory where rpmbuild will put the RPM
+		//  - --buildroot is the work directory where rpmbuild will build the RPM
+		//  - -ba tells rpmbuild to build the source and binary RPMs
+		//
+		// TODO(cpuguy83): specPath would ideally never change.
+		// We don't want to have to re-run this step just because the path changed, this should be based on inputs only (ie the content of the rpm spec, sources, etc)
```

Comment:
How are you thinking of making this work without the `specPath` as a parameter? Pass the spec through inline? Or maybe if we are only ever building one SPEC at once (possible asterisk) maybe we could hardcode the SPEC path for this step and then copy the SPEC somewhere else later?

Reply:
I'm not sure right now, which is why I added the todo.
In practice I doubt this will even be an issue.